bson/json_util.py

```diff
@@ -486,13 +492,13 @@ def _get_dbpointer(doc, dummy):
 
 
 def canonical_object_hook(dct, json_options=CANONICAL_JSON_OPTIONS):
-    keyset = frozenset(dct)
+    keyset = frozenset(key for key in dct if key.startswith('$'))
```

Comment:
It is parsed into an int32. Because the extra keys do not start with `$`, they don't "count against" the type of the document. Here's a test that demonstrates the expected behavior with DBPointer: https://github.com/mongodb/specifications/blob/6cee58a86008ed3ee4586825ac0acfeab8b5263b/source/bson-corpus/tests/dbpointer.json#L18-L23. Note that this behavior is consistent json_util's non-canonical behavior with `object_hook`, e.g., if we see `$oid` in a dictionary, interpret the `$oid` field as an ObjectId hex string and forget about any other keys.